### PR TITLE
Dedupe yarn.lock

### PR DESCRIPTION
**Pull request by FrederikBolding**
Ran `yarn dedupe` to remove duplicates from our `yarn.lock`.

This reduced the size of `node_modules` from 582 MB to 455 MB (-21% diff!).

This PR also adds a check to CI to enforce this from now on.

### CODE REVIEW

File: .github/workflows/build-lint-test.yml [hunk 1]
```diff
@@ -27,6 +27,7 @@ jobs:
           path: .yarn/cache
           key: yarn-cache-${{ runner.os }}-${{ steps.yarn-version.outputs.YARN_VERSION }}-${{ hashFiles('yarn.lock') }}
       - run: yarn install --immutable
+      - run: yarn dedupe --check
```
Comment (ritave) [inline .github/workflows/build-lint-test.yml:30]: The CI check might become annoying quickly, and it's not really an error. Git hook might be a better placement

File: packages/utils/src/json-schemas/validateNpmSnapPackageJson.js [hunk 2]
```diff
@@ -1 +1 @@
...
...l){vErrors = [err21];}else {vErrors.push(err21);}errors++;}}if(data7.url !== undefined){let data9 = data7.url;if(typeof data9 === "string"){if(func4(data9) < 1){const err22 = {instancePath:instancePath+"/repository/url",schemaPath:"#/properties/repository/properties/url/minLength",keyword:"minLength",params:{limit: 1},message:"must NOT have fewer than 1 characters"};if(vErrors === null){vErrors = [err22];}else {vErrors.push(err22);}errors++;}}else {const err23 = {instancePath:instancePath+"/repository/url",schemaPath:"#/properties/repository/properties/url/type",keyword:"type",params:{type: "string"},message:"must be string"};if(vErrors === null){vErrors = [err23];}else {vErrors.push(err23);}errors++;}}}else {const err24 = {instancePath:instancePath+"/repository",schemaPath:"#/properties/repository/type",keyword:"type",params:{type: "object"},message:"must be object"};if(vErrors === null){vErrors = [err24];}else {vErrors.push(err24);}errors++;}}}else {const err25 = {instancePath,schemaPath:"#/type",keyword:"type",params:{type: "object"},message:"must be object"};if(vErrors === null){vErrors = [err25];}else {vErrors.push(err25);}errors++;}validate20.errors = vErrors;return vErrors;}
```
Comment (ritave) [inline packages/utils/src/json-schemas/validateNpmSnapPackageJson.js:1]: Why the changes in those files?

**Review by ritave — CHANGES_REQUESTED** [review]
The CI check might become annoying quickly, and it's not really an error. Git hook might be a better placement

File: .github/workflows/build-lint-test.yml (hunk 1 above)
Comment (FrederikBolding) [reply .github/workflows/build-lint-test.yml:30]: What is the scenario where you imagine it being annoying? 🤔

Comment (ritave) [reply .github/workflows/build-lint-test.yml:30]: Anytime I update a dependency my CI might become rejected, I now how to go back and check the logs, make a new commit, push and wait again.

Making this optional but automatic in a git hook would manage the size automatically and make for a nicer user experience

Comment (FrederikBolding) [reply .github/workflows/build-lint-test.yml]: Done!

**Review by ritave — APPROVED** [review]
